src/components/legend.ts

```diff
@@ -1,4 +1,233 @@
 namespace Plottable.Components {
+  /**
+   * The Legend's column representation. Stores position information
+   * as well as data
+   */
+  class LegendColumn<T> {
+    constructor(private _width = 0, private _height = 0, private _data: T) {}
+
+    /**
+     * @return {T} data Returns any data stored in the column
+     */
+    public getData() {
+      return this._data;
+    }
+
+    /**
+     * @return {number} height The height of the column.
+     */
+    public getHeight() {
+      return this._height;
+    }
+
+    /**
+     * @return {number} width The width of the column.
+     */
+    public getWidth() {
+      return this._width;
+    }
+
+    /**
+     * @param {number} width Sets the width of the column
+     */
+    public setWidth(width: number) {
+      return this._width = width;
+    }
+  }
+
+  /**
+   * The Legend's row representations. Stores positioning information
+   * and column data.
+   */
+  class LegendRow {
```

Comment:
Will "row" and "column" always make sense? Will we ever rotate this text? Will we ever present non-grid-aligned (what about center aligned) legends?